internal/app/networkd/pkg/nic/netlink.go

```diff
@@ -0,0 +1,128 @@
+package nic
+
+import (
+	"log"
+	"net"
+
+	"github.com/jsimonetti/rtnetlink"
+	"github.com/mdlayher/netlink"
+	"golang.org/x/sys/unix"
+)
+
+// createLink creates an interface
+func (n *NetworkInterface) createLink() error {
+	var info *rtnetlink.LinkInfo
+
+	if n.Bonded {
+		info = &rtnetlink.LinkInfo{Kind: "bond"}
+	}
+
+	err := n.rtConn.Link.New(&rtnetlink.LinkMessage{
+		Family: unix.AF_UNSPEC,
+		Type:   0,
+		Attributes: &rtnetlink.LinkAttributes{
+			Name: n.Name,
+			Info: info,
+		},
+	})
+
+	return err
+}
+
+// setMTU sets the link MTU
+func (n *NetworkInterface) setMTU(idx int, mtu uint32) error {
+	msg, err := n.rtConn.Link.Get(uint32(idx))
+	if err != nil {
+		log.Printf("failed to get link %d\n", idx)
+		return err
+	}
+
+	err = n.rtConn.Link.Set(&rtnetlink.LinkMessage{
+		Family: msg.Family,
+		Type:   msg.Type,
+		Index:  uint32(idx),
+		Flags:  msg.Flags,
+		Change: 0,
+		Attributes: &rtnetlink.LinkAttributes{
+			MTU: mtu,
+		},
+	})
+
+	return err
+}
+
+// nolint: unused
+func (n *NetworkInterface) configureBond(idx int, attrs []netlink.Attribute) error {
+	log.Println("configuring bond")
```

Comment:
Since this is a lower-level library, I think we should keep the logging out of it, and bubble it up as far as possible. I envision us pulling this out one day in a package that anyone can use.